### ISSUE TITLE
Question about some details of training

### ISSUE THREAD

**Issue by connorye** [om-ai-lab/VLM-R1#298]
I try to finetune on omlab/Qwen2.5VL-3B-VLM-R1-REC-500steps based on the readme with my own custom dataset,which has the same format of refcoco.And it seemed weired why the loss is 0.0?


Parameter Offload: Total persistent parameters: 755712 in 408 params
  0%|                                                                                                                                                                                                                                                            | 0/7280 [00:00<?, ?it/s]/home/userH/anaconda3/envs/vr/lib/python3.12/site-packages/torch/utils/checkpoint.py:87: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn(
/home/userH/anaconda3/envs/vr/lib/python3.12/site-packages/torch/utils/checkpoint.py:87: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn(
/home/userH/anaconda3/envs/vr/lib/python3.12/site-packages/torch/utils/checkpoint.py:87: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn(
/home/userH/anaconda3/envs/vr/lib/python3.12/site-packages/torch/utils/checkpoint.py:87: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn(
`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`...
DynamicCache + torch.export is tested on torch 2.6.0+ and may not work on earlier versions.
`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`...
DynamicCache + torch.export is tested on torch 2.6.0+ and may not work on earlier versions.
`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`...
DynamicCache + torch.export is tested on torch 2.6.0+ and may not work on earlier versions.
`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`...
DynamicCache + torch.export is tested on torch 2.6.0+ and may not work on earlier versions.
[2025-06-10 18:16:05,528] [WARNING] [stage3.py:2150:step] 2 pytorch allocator cache flushes since last step. this happens when there is high memory pressure and is detrimental to performance. if this is happening frequently consider adjusting settings to reduce memory consumption. If you are unable to make the cache flushes go away consider adding get_accelerator().empty_cache() calls in your training loop to ensure that all ranks flush their caches at the same time
{'loss': 0.0, 'grad_norm': 3.6189626616779864, 'learning_rate': 1e-06, 'completion_length': 128.8125, 'rewards/iou_reward': 0.014366726391017437, 'rewards/format_reward_rec': 1.0, 'reward': 1.0143667459487915, 'reward_std': 0.027531811967492104, 'kl': 0.0, 'clip_ratio': 0.0, 'epoch': 0.0}
  0%|                                                                                                                                                                                                                                                | 1/7280 [01:08<137:32:26, 68.02s/it][2025-06-10 18:17:14,457] [WARNING] [stage3.py:2150:step] 1 pytorch allocator cache flushes since last step. this happens when there is high memory pressure and is detrimental to performance. if this is happening frequently consider adjusting settings to reduce memory consumption. If you are unable to make the cache flushes go away consider adding get_accelerator().empty_cache() calls in your training loop to ensure that all ranks flush their caches at the same time
{'loss': 0.0, 'grad_norm': 2.184789171159602, 'learning_rate': 9.998626373626373e-07, 'completion_length': 141.375, 'rewards/iou_reward': 0.015515854116529226, 'rewards/format_reward_rec': 0.984375, 'reward': 0.9998908638954163, 'reward_std': 0.06634941324591637, 'kl': 0.000457763671875, 'clip_ratio': 0.0, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 2.4564408810295353, 'learning_rate': 9.997252747252746e-07, 'completion_length': 125.546875, 'rewards/iou_reward': 0.0018199582700617611, 'rewards/format_reward_rec': 0.984375, 'reward': 0.9861949682235718, 'reward_std': 0.04852189589291811, 'kl': 0.00046634674072265625, 'clip_ratio': 0.0, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 6.836600933150137, 'learning_rate': 9.99587912087912e-07, 'completion_length': 136.96875, 'rewards/iou_reward': 0.0052139414474368095, 'rewards/format_reward_rec': 1.0, 'reward': 1.0052139163017273, 'reward_std': 0.01015559770166874, 'kl': 0.0005779266357421875, 'clip_ratio': 0.0, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 1.734693460724659, 'learning_rate': 9.994505494505493e-07, 'completion_length': 134.125, 'rewards/iou_reward': 0.006291818805038929, 'rewards/format_reward_rec': 1.0, 'reward': 1.0062918663024902, 'reward_std': 0.013452197890728712, 'kl': 0.0006008148193359375, 'clip_ratio': 0.0, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 3.3101529415169453, 'learning_rate': 9.993131868131869e-07, 'completion_length': 135.96875, 'rewards/iou_reward': 0.011188212083652616, 'rewards/format_reward_rec': 1.0, 'reward': 1.011188268661499, 'reward_std': 0.021582315210253, 'kl': 0.001064300537109375, 'clip_ratio': 0.0, 'epoch': 0.0}
  0%|▏

**Comment by VincentYCYao** [comment]
I also got `UserWarning: None of the inputs have requires_grad=True. Gradients will be None`

**Comment by liyyyyyyyyyyyyy** [comment]
> 问题解决了吗？我也遇到了同样的问题。

请问您解决了这个问题了吗

**Comment by liyyyyyyyyyyyyy** [comment]
> 我还得到了`UserWarning: None of the inputs have requires_grad=True. Gradients will be None`

请问您解决了这个问题了吗

**Comment by liyyyyyyyyyyyyy** [comment]
> 我尝试在omlab/Qwen2.5VL-3B-VLM-R1-REC-500steps上根据说明文件和我自己的自定义数据集微调，这个数据集格式和refcoco格式一样。而且看起来很奇怪为什么损失是0.0？
> 
> 参数卸载：总持久参数：755712 408个参数 0%| |0/7280 [00：00<?, ?it/s]/home/userH/anaconda3/envs/vr/lib/python3.12/site-packages/torch/utils/checkpoint.py：87： UserWarning：所有输入均无 requires_grad=True。梯度将是 None warnings.warn（ /home/userH/anaconda3/envs/vr/lib/python3.12/site-packages/torch/utils/checkpoint.py：87： UserWarning： 所有输入都没有 requires_grad=True。梯度将是 None warnings.warn（ /home/userH/anaconda3/envs/vr/lib/python3.12/site-packages/torch/utils/checkpoint.py：87： UserWarning：所有输入都没有 requires_grad=True。梯度将是 无 warnings.warn（ /home/userH/anaconda3/envs/vr/lib/python3.12/site-packages/torch/utils/checkpoint.py：87： UserWarning： 所有输入都没有 requires_grad=True。梯度将显示为无警告。警告（ 与梯度检查点不兼容。设置...... DynamicCache + torch.export 在 torch 2.6.0+ 上测试，可能无法在早期版本上运行。与梯度检查点不兼容。设置...... DynamicCache + torch.export 在 torch 2.6.0+ 上测试，可能无法在早期版本上运行。与梯度检查点不兼容。设置...... DynamicCache + torch.export 在 torch 2.6.0+ 上测试，可能无法在早期版本上运行。与梯度检查点不兼容。设置...... DynamicCache + torch.export 在 torch 2.6.0+ 上测试，可能无法在早期版本上运行。 [2025-06-10 18：16：05,528] [警告] [stage3.py：2150：step] 自上一步以来，Pytorch分配缓存已清除两次。这通常发生在记忆压力较大时，对表现有害。如果这种情况频繁发生，可以考虑调整设置以减少内存消耗。如果无法消除缓存冲洗，建议在训练循环中添加get_accelerator（）.empty_cache（）调用，确保所有等级同时清空缓存 {'loss'： 0.0， 'grad_norm'： 3.6189626616779864， 'learning_rate'： 1e-06， 'completion_length'： 128.8125， 'rewards/iou_reward'： 0.014366726391017437， 'rewards/format_reward_rec'： 1.0， 'reward'： 1.0143667459487915， 'reward_std'： 0.027531811967492104， 'kl'： 0.0， 'clip_ratio'： 0.0， 'epoch'： 0.0} 0%| |1/7280 [01：08<137：32：26， 68.02s/it][2025-06-10 18：17：14,457] [警告] [stage3.py：2150：step] 自上次步骤以来，1次pytorch分配器缓存清除。这通常发生在记忆压力较大时，对表现有害。如果这种情况频繁发生，可以考虑调整设置以减少内存消耗。如果你无法让缓存冲刷消失，可以考虑在训练循环中添加get_accelerator（）.empty_cache（）呼叫，确保所有等级同时清空缓存 {'loss'： 0.0， 'grad_norm'： 2.184789171159602， 'learning_rate'： 9.9986263736266373e-07， 'completion_length'： 141.375， 'rewards/iou_reward'： 0.015515854116529226， 'rewards/format_reward_rec'： 0.984375， 'reward'： 0.9998908638954163， 'reward_std'：0.06634941324591637，'kl'：0.000457763671875，'clip_ratio'：0.0，'epoch'：0.0} {'loss'：0.0，'grad_norm'：2.4564408810295353，'learning_rate'：9.99725274725272746e-07，'completion_length'：125.546875，'rewards/iou_reward'：0.0018199582700617611，'rewards/format_reward_rec'：0.984375，'reward'：0.9861949682235718，'reward_std'：0.04852189589291811，'kl'：0.00046634674072265625，'clip_ratio'：0。 0， 'epoch'： 0.0} {'loss'： 0.0， 'grad_norm'： 6.83600933150137， 'learning_rate'： 9.99587912087912087912e-07， 'completion_length'： 136.96875， 'Rewards/iou_reward'： 0.0052139414474368095， 'rewards/format_reward_rec'： 1.0， 'reward'： 1.0， 'reward'： 1.0052139163017273， 'reward_std'： 0.01015559770166874， 'kl'： 0.0005779266357421875， 'clip_ratio'： 0.0， 'epoch'： 0.0}{'损失'：0.0，'grad_norm'：1.734693460724659，'learning_rate'：9.994505494505493e-07，'completion_length'：134.125，'奖励/iou_reward'：0.0062918188805038929，'奖励/format_reward_rec'：1.0，'奖励'：1.0062918663024902，'reward_std'：0.013452197890728712，'kl'：0.0006008148193359375，'clip_ratio'：0.0，'epoch'：0.0} {'损失'： 0.0， 'grad_norm'： 3.3101529415169453， 'learning_rate'： 9.993131868131869e-07， 'completion_length'： 135.96875， '奖励/iou_reward'： 0.011188212083652616， 'rewards/format_reward_rec'： 1.0， 'reward'： 1.0118826661499， 'reward_std'： 0.021582315210253， 'kl'： 0.001064300537109375， 'clip_ratio'： 0.0， 'epoch'： 0.0} 0%|▏`use_cache=True``use_cache=False``use_cache=True``use_cache=False``use_cache=True``use_cache=False``use_cache=True``use_cache=False`

请问您解决这个问题了吗
